dll/ntdll/ldr/ldrutils.c

```diff
@@ -1977,8 +2000,10 @@ LdrpCheckForLoadedDll(IN PWSTR DllPath,
 
     /* Look in the hash table if flag was set */
 lookinhash:
-    if (Flag)
+    if (Flag  /* the second check is a hack */ && !RedirectedDll)
     {
+        /* FIXME: if we get redirected dll it means that we also get a full path so we need to find its filename for the hash lookup */
```

Comment:
I don't see how that's a problem. What does "flag" mean in this context ? From the code below, that looks like "not a full path"

Reply:
The thing is that at this point we should do a fast lookup uing the hash, We can do that for redirected dlls as well . However the hash is computed only based on the filename so I was lazy making it find the filename when I wrote  this.